src/core/src/test/scala/play/api/db/slick/DefaultSlickApiSpec.scala

```diff
@@ -24,31 +24,36 @@ class DefaultSlickApiSpec extends Specification with Mockito { self =>
   }
 
   "DefaultSlickApi" should {
+
     "check the assumption that the ApplicationLifecycle isn't null" in {
       val lifecycle = injector.instanceOf[ApplicationLifecycle]
       lifecycle must not(beNull)
     }
+
     "check the assumption that the ApplicationLifecycle is a singleton" in {
       val injector = self.injector
       val lifecycle1 = injector.instanceOf[ApplicationLifecycle]
       val lifecycle2 = injector.instanceOf[ApplicationLifecycle]
       lifecycle1 mustEqual lifecycle2
     }
+
     "check the assumption that ApplicationLifecycle is binded to a DefaultApplicationLifecycle" in {
       val lifecycle = injector.instanceOf[ApplicationLifecycle]
       lifecycle must beAnInstanceOf[DefaultApplicationLifecycle]
     }
-    "check that no stop hook is registered in the ApplicationLifecycle at application start" in {
-      val lifecycle = injector.instanceOf[ApplicationLifecycle].asInstanceOf[DefaultApplicationLifecycle]
-      hooks(lifecycle) must have size (0)
-    }
```

Comment:
this test is not adding any value to play-slick. Why would we test that there is not hook added by Play?

What we need to confirm is that `SlickApi` is adding its own hook. That's all.
See next test for more context.